link/query_test.go

```diff
@@ -0,0 +1,84 @@
+package link
+
+import (
+	"testing"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal/testutils"
+)
+
+func TestQueryPrograms(t *testing.T) {
+	for name, fn := range map[string]func(*testing.T) (*ebpf.Program, QueryOptions){
+		"cgroup": queryCgroupFixtures,
+		"netns":  queryNetNSFixtures,
+	} {
+		t.Run(name, func(t *testing.T) {
+			prog, opts := fn(t)
+			ids, err := QueryPrograms(opts)
+			testutils.SkipIfNotSupported(t, err)
+			if err != nil {
+				t.Fatal("Can't query programs:", err)
+			}
+
+			progInfo, err := prog.Info()
+			if err != nil {
+				t.Fatal("Can't get program info:", err)
+			}
+
+			progId, ok := progInfo.ID()
+			if !ok {
+				t.Skip("Program ID not supported")
```

Comment:
Nit: doesn't PROG_QUERY imply IDs being present? Probably fine to bail out here.

Reply:
It does indeed. Should I just not check the `ok` var here? `ID` is available from 4.13 and we skip this test until 4.15 with the feature test anyway.

Reply:
yes, drop the check